idea/src/org/jetbrains/kotlin/idea/inspections/ConstantConditionIfInspection.kt

```diff
@@ -157,27 +159,59 @@ private fun KtExpression.enumEntry(): KtEnumEntry? {
 }
 
 fun KtExpression.replaceWithBranch(branch: KtExpression, isUsedAsExpression: Boolean, keepBraces: Boolean = false) {
-    val lastExpression = when {
-        branch !is KtBlockExpression -> replaced(branch)
+    val caretModel = findExistingEditor()?.caretModel
+
+    val subjectVariable = (this as? KtWhenExpression)?.subjectVariable?.let { it ->
+        if (it.annotationEntries.isNotEmpty()) return@let it
+        val initializer = it.initializer ?: return@let it
+        val references = ReferencesSearch.search(it, LocalSearchScope(this)).toList()
+        if (references.size > 1) return@let it
+        references.firstOrNull()?.element?.replace(initializer)
+        null
+    }
+
+    val factory = KtPsiFactory(this)
+    val parent = this.parent
+    val replaced = when {
+        branch !is KtBlockExpression -> {
+            if (subjectVariable != null) {
+                if (isUsedAsExpression) {
+                    replaced(KtPsiFactory(this).createExpressionByPattern("run { $0\n$1 }", subjectVariable, branch))
+                } else {
```

Comment:
I think we shouldn't expand the scope for subjectVariable.
For example:
``` Kotlin
when (val a = 42) {
    else -> use(a, a)
}

val a = 33
```

Reply:
I want to return null if references size is 0.